tests/integration/cheatcodes/deploy_contract/deploy_contract_test.cairo

```diff
@@ -16,96 +16,202 @@ namespace ProxyContract:
-
-    ProxyContract.set_target(
-        contract_address=contract_proxy_address, new_target=contract_logic_address)
+@external
+func test_deploy_contract_with_contructor{syscall_ptr : felt*, range_check_ptr}():
+    alloc_locals
+    local contract_address : felt
+    %{
+        ids.contract_address = deploy_contract("./tests/integration/cheatcodes/deploy_contract/basic_with_constructor.cairo", [41]).contract_address
+    %}
+    BasicWithConstructor.increase_balance(contract_address, 1)
+    let (res) = BasicWithConstructor.get_balance(contract_address)
+    assert res = 42
+    return ()
+end
 
-    ProxyContract.increase_twice(contract_address=contract_proxy_address, amount=5)
+@external
+func test_deploy_contract_with_contructor_steps{syscall_ptr : felt*, range_check_ptr}():
+    alloc_locals
+    local contract_address : felt
+    %{
+        declared = declare("./tests/integration/cheatcodes/deploy_contract/basic_with_constructor.cairo")
+        prepared = prepare(declared, [41])
+        ids.contract_address = deploy(prepared).contract_address
+    %}
+    BasicWithConstructor.increase_balance(contract_address, 1)
+    let (res) = BasicWithConstructor.get_balance(contract_address)
+    assert res = 42
     return ()
 end
 
+
 @external
-func test_passing_constructor_data_as_list{syscall_ptr : felt*, range_check_ptr}():
+func test_deploy_contract_pranked{syscall_ptr : felt*, range_check_ptr}():
     alloc_locals
-    local deployed_contract_address : felt
-    let (contract_address) = get_contract_address()
 
     %{
-        ids.deployed_contract_address = deploy_contract("./tests/integration/cheatcodes/deploy_contract/basic_with_constructor.cairo",
-            [42, 0, ids.contract_address]
-        ).contract_address
+        declared_1 = declare("./tests/integration/cheatcodes/deploy_contract/pranked_contract.cairo")
+        declared_2 = declare("./tests/integration/cheatcodes/deploy_contract/pranked_contract.cairo")
+
+        prepared_1 = prepare(declared_1, [111])
+        prepared_2 = prepare(declared_2, [222])
+
+        start_prank(111, target_contract_address=prepared_1.contract_address)
+        start_prank(222, target_contract_address=prepared_2.contract_address)
+
+        deploy(prepared_1)
+        deploy(prepared_2)
     %}
+    return ()
+end
 
-    let (balance) = BasicWithConstructor.get_balance(deployed_contract_address)
-    let (id) = BasicWithConstructor.get_id(deployed_contract_address)
+@external
+func test_deploy_the_same_contract_twice{syscall_ptr : felt*, range_check_ptr}():
+    %{
+        declared_1 = declare("./tests/integration/cheatcodes/deploy_contract/basic_with_constructor.cairo")
+        declared_2 = declare("./tests/integration/cheatcodes/deploy_contract/basic_with_constructor.cairo")
 
-    assert balance.low = 42
-    assert balance.high = 0
-    assert id = contract_address
+        prepared_1 = prepare(declared_1, [42])
+        prepared_2 = prepare(declared_2, [42])
```

Comment:
What does happen if somebody tries deploying the same declared contract twice with different args?

Reply:
Why are we not using chaining `declare().prepare(CONSTRUCTOR_ARGS).deploy()`? (There was some reason, but I've forgotten.) We could support skipping prepare step nicely. `declare().deploy(CONSTRUCTOR_ARGS)`